app/models/cloud_volume/operations.rb

```diff
@@ -28,6 +28,28 @@ def attach_volume(server_ems_ref, device = nil)
     raw_attach_volume(server_ems_ref, device)
   end
 
+  def clone_volume_queue(userid, options = {})
+  task_opts = {
+    :action => "cloning Cloud Volume for user #{userid}",
+    :userid => userid
+  }
+
+  queue_opts = {
+    :class_name  => self.class.name,
+    :method_name => 'clone_volume',
+    :instance_id => id,
+    :role        => 'ems_operations',
+    :queue_name  => ext_management_system.queue_name_for_ems_operations,
+    :zone        => ext_management_system.my_zone,
+    :args        => [options]
+  }
+  MiqTask.generic_action_with_callback(task_opts, queue_opts)
```

Comment:
@jdenver23 looks like we still need to fix the indentation level here,

Reply:
My apologies, thought I had gotten that!